Assignment3/PointAndGrid.py

```diff
@@ -0,0 +1,260 @@
+class Point:
+
+    def __init__(self, grid = None, coordinates = None, value = None):
+        """
+        For Point initialization with optional grid and coordinates.
+        :param grid: Grid
+        :param coordinates: Tuple(int, int)
+        :return: None
+        """
+        self.grid = grid
+        self.coordinates = coordinates
+        self.value = value
+        if grid and coordinates:
+            v = grid.getElement(coordinates)
+            if value and value != v:
+                raise AttributeError
+            else: self.value = v
+
+    def setGrid(self, gr):

```

Comment:
Since `self.grid`, `self.coordinates` and `self.value` are all public attributes (by naming convention) I would suggest removing these methods and setting the instance attributes directly. For "pythonic" setters/getters with custom implementations, take a look at `@property` decorator:

https://docs.python.org/3.6/library/functions.html#property